internal/rpc/rpcserver/server.go

```diff
@@ -2430,10 +2433,28 @@ func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr
 		}
 	}
 
+	// new vspd request
+	var vspHost string
+	var vspPubKey string
+	if req.VspHost != "" || req.VspPubkey != "" {
+		vspHost = req.VspHost
+		vspPubKey = req.VspPubkey
+		if vspPubKey == "" {
+			return status.Errorf(codes.InvalidArgument, "vsp pubkey can not be null")
+		}
+		if vspHost == "" {
+			return status.Errorf(codes.InvalidArgument, "vsp host can not be null")
+		}
+	}
```

Comment:
we should error if the request contains both legacy stakepoold and vspd options.  Both cannot be set.

Reply:
True. Added a check for that